src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_fankriss.cpp

```diff
@@ -15,202 +15,175 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-/* ScriptData
-SDName: Boss_Fankriss
-SD%Complete: 100
-SDComment: sound not implemented
-SDCategory: Temple of Ahn'Qiraj
-EndScriptData */
+/*
+ * Timers requires to be revisited
+ */
 
 #include "ScriptMgr.h"
+#include "Containers.h"
 #include "ScriptedCreature.h"
+#include "SpellInfo.h"
+#include "SpellScript.h"
 #include "temple_of_ahnqiraj.h"
-#include "TemporarySummon.h"
 
-#define SOUND_SENTENCE_YOU 8588
-#define SOUND_SERVE_TO     8589
-#define SOUND_LAWS         8590
-#define SOUND_TRESPASS     8591
-#define SOUND_WILL_BE      8592
-
-enum Spells
+enum FankrissSpells
 {
-    SPELL_MORTAL_WOUND      = 28467,
-    SPELL_ROOT              = 28858,
+    SPELL_MORTAL_WOUND      = 25646,
+
+    SPELL_SUMMON_WORM_1     = 518,
+    SPELL_SUMMON_WORM_2     = 25831,
+    SPELL_SUMMON_WORM_3     = 25832,
 
-    // Enrage for his spawns
-    SPELL_ENRAGE            = 28798
+    SPELL_ENTANGLE_1        = 720,
+    SPELL_ENTANGLE_2        = 731,
+    SPELL_ENTANGLE_3        = 1121,
+
+    SPELL_SPAWN_HATCHLING_1 = 26630,
+    SPELL_SPAWN_HATCHLING_2 = 26631,
+    SPELL_SPAWN_HATCHLING_3 = 26632
 };
 
-class boss_fankriss : public CreatureScript
+enum FankrissEvents
 {
-public:
-    boss_fankriss() : CreatureScript("boss_fankriss") { }
+    EVENT_MORTAL_WOUND      = 1,
+    EVENT_SUMMON_WORM,
+    EVENT_ENTANGLE
+};
 
-    CreatureAI* GetAI(Creature* creature) const override
+std::vector<uint32> SummonWormSpells = { SPELL_SUMMON_WORM_1, SPELL_SUMMON_WORM_2, SPELL_SUMMON_WORM_3 };
+std::vector<uint32> EntangleSpells = { SPELL_ENTANGLE_1, SPELL_ENTANGLE_2, SPELL_ENTANGLE_3 };
+
+// 15510 - Fankriss the Unyielding
+struct boss_fankriss : public BossAI
+{
+    boss_fankriss(Creature* creature) : BossAI(creature, DATA_FRANKRIS), _wormsSpawnPerWave(0), _entanglePerWave(0) { }
+
+    void Reset() override
     {
-        return GetAQ40AI<boss_fankrissAI>(creature);
+        _Reset();
+        _wormsSpawnPerWave = urand(1, 3);
+        Trinity::Containers::RandomShuffle(SummonWormSpells);
```

Comment:
you are shuffling it for -every active script instance- aka globally here. That means:
if i'm infight with him solo and you are resetting him in a different instance it'll get shuffled for me aswell. I dont think this is intended. You would need a local copy within this script to prevent it.
e.g. by defining the "spells in general" as an array and then creating an vector in this script based on the array (e.g. like in rogue roll_the_bones script in master branch) and you just shuffle the vector then. That means only the vector for this particular script is shuffled, not globally.

Reply:
Done